File: AOJ/0299/railroad2.py
dst = []

def CWDist(s, e, n) : return (e - s + n) % n
def CCWDist(s, e, n) : return (s - e + n) % n
def CWNext(i, m) : return dst[0] if i == m - 1 else dst[i + 1]
def CCWNext(i, m) : return dst[m-1] if i == 0 else dst[i - 1]
def CWTurn(p, t, e, n) : return CWDist(p, t, n) + CCWDist(t, e, n)
def CCWTurn(p, t, e, n) : return CCWDist(p, t, n) + CWDist(t, e, n)

n,m,p = map(int, input().split())
for i in range(m) :
    dst.append(int(input()))
dst.sort()

s = 0
while s < m :
    if dst[s] > p : break
    s += 1

cwNext = dst[0] if s == m else dst[s]
ccwNext = dst[m - 1] if s == 0 else dst[s - 1]
dist = min(CCWDist(p, cwNext, n), CWDist(p, ccwNext, n))
for i in range(m) :
    dist = min(dist, CWTurn(p, dst[i], CWNext(i,m), n), CCWTurn(p, dst[i], CCWNext(i,m), n))
print(100 * dist)

File: AOJ/0299/railroad2.cpp
#include<iostream>
#include<algorithm>
#include<vector>
using namespace std;

int n, m, p;
vector<int> dst;

int CWDist (int s, int e) { return (e - s + n) % n; }
int CCWDist (int s, int e) { return (s - e + n) % n; }

int CWNext ( int i ) { return i == m - 1 ? dst[0] : dst[i + 1]; }
int CCWNext ( int i ) { return i == 0 ? dst[m - 1] : dst[i - 1]; }

int CWTurn ( int p, int t, int e ) { return CWDist(p, t) + CCWDist(t, e); }
int CCWTurn ( int p, int t, int e ) { return CCWDist(p, t) + CWDist(t, e); }

int main() {
  cin >> n >> m >> p;
  dst.resize( m );
    
  for ( int i = 0; i < m; ++i ) cin >> dst[i];
  sort(dst.begin(), dst.end());

  int s = 0;
  while ( s < m ) {
    if ( dst[s] > p ) break;
    ++s;
  }
    
  int cwNext  = s == m ? dst[0] : dst[s];
  int ccwNext = s == 0 ? dst[m - 1] : dst[s - 1];
  int dist = min(CCWDist(p, cwNext), CWDist(p, ccwNext));

  for ( int i = 0; i < m; ++i ) {
    dist = min(dist, CWTurn(p, dst[i], CWNext(i)));
    dist = min(dist, CCWTurn(p, dst[i], CCWNext(i)));
  }

  cout << 100 * dist << endl;
  return 0;
}
